big-int/src/operators/greater_than.cpp

```diff
@@ -32,40 +32,51 @@
 
 namespace libbig
 {
+//!  Operator overload for > for largeInt class
+/*!
+    This operator returns a bool value which is 
+    1 when the > condition is satisfied between the two largeInt objects
+    0 otherwise
+*/
 bool largeInt::operator>(const largeInt &a)
 {
+    /** If first number is negative and second number is positive 
+     *  then > condition is not satisfied.
+     */
     if(this->sign == NEGATIVE && a.sign == POSITIVE)
     {
         return 0;
     }
+    /** If first number is positive and second number is negative 
+     *  then > condition is satisfied.
+     */
     if(a.sign == NEGATIVE && this->sign == POSITIVE)
     {
         return 1;
     }
+    /** If both the numbers have same sign 
+     *  then comparing their lengths
+     */
     if(this->number.length() > a.number.length())
     {
-        return 0^this->sign;
-    }
+        return 0^this->sign;    //! if first number is longer and both are negative then the condition is satisfied.
+    }                           //! but if first number is longer and both are positive then the condition is not satisfied.
     if(this->number.length() < a.number.length())
     {
-        return 1^this->sign;
-    }
+        return 1^this->sign;    //! if first number is shorter and both are negative then the condition is not satisfied.
+    }                           //! but if first number is shorter and both are positive then the condition is satisfied.
 
-    for(int i=0;i<a.number.length();i++)
+    for(int i=0;i<a.number.length();i++)    //! if the length of both numbers are same then we compare each digit from LTR and the first non equal digit can determine whether the condition is satisfied
```

Comment:
just a minor suggestion, it's a good practice followed by many good organisations to not to exceede a line more than 80 characters. Basically by doing so we get rid of the scrolling issue and it becomes easy to work with multiple files of the code base on a single screen.
```C++
 /**
  * if the length of both numbers are same then we compare each digit from LTR
  * and the first non equal digit can determine whether the condition is satisfied
  */
  for(int i=0;i<a.number.length();i++) 
```

for reference you can look at these files
https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/graph/breadth_first_search.cpp
https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/graph/dijkstra.cpp
https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/graph/depth_first_search.cpp

Reply:
doxygen also suppors markdown, HTML, LaTex support, you can use any of these things for better formatting

Reply:
Since the library is currently in its initial stage, I propose we should focus more on developing the code instead. The documentation part could be completed later. It can also be a good issue for someone. As long as the basic functionalities of the library are not implemented, further development is not possible.

Reply:
I think aspect of documentation is as important as that of code because if the documentation is not proper there is hardly someone who will prefer to use a such library. There are a lot of libraries which are good enough but lack in documentation aspect and thus people end up leaving them aside because most of them find hard to start with them.
And also i think the person who is implementing some feature X will be able to write about it more clearly. Therefore leaving the documentation for someone else is not as effective as this is.

Well i am not enforcing constraints over you, i had given just some suggestions.
And for focusing on code i am creating issues one by one, with a hope that someone will pick up and start with them soon.

Reply:
I get your point. It's true that a code without proper documentation isn't very useful for others. The points and resources that you mentioned are really helpful and something that I was not aware of before. I really thank you for that.

Actually, I am having less time these days due to my academic workload that is why I couldn't get enough time to write a proper documentation. I wish to add the code I have written for this repository so far and I promise to add proper documentation for the codes I wrote as soon as I get time. :)

Reply:
@yashsoni501 no problem you can mark this PR as draft, whenever you feel it's ready for review just ping me.

Reply:
@ayaankhan98 Can you merge these changes in the repository? This will save someone's time to go through writing the code again for something which is already implemented. Also my other codes for other operators which are almost ready depend on these basic operators.